testutil/daemon/daemon.go

```diff
@@ -842,6 +843,17 @@ func (d *Daemon) ReloadConfig() error {
 	return nil
 }
 
+// SetEnvVar updates the set of extra env variables for the daemon, to take
+// effect on the next start/restart.
+func (d *Daemon) SetEnvVar(name, val string) {
+	prefix := name + "="
+	if idx := slices.IndexFunc(d.extraEnv, func(ev string) bool { return strings.HasPrefix(ev, prefix) }); idx > 0 {
+		d.extraEnv[idx] = prefix + val
+		return
+	}
+	d.extraEnv = append(d.extraEnv, prefix+val)
+}
```

Comment:
This looks sane for now, but I went looking "didn't we already  have an options for this?", but that can only be used on `daemon.New()`;

```go
d := daemon.New(t, daemon.WithEnvVars(...)
```


and `daemon.Start()` doesn't take options as argument; perhaps we should consider making it accept Options, so that we can customise (some at least) daemon settings between restarts without having to create a new daemon.

Or perhaps a `StartWithOptions()` if we don't want  to replace the existing one (which takes  command-line arguments only)